crates/source_analyzer/src/structs.rs

```diff
@@ -304,30 +178,29 @@ fn path_to_insertion_rules(path: &[PathSegment], data_type: &Type) -> Vec<Insert
         .chain(once(InsertionRule::AppendDataType {
             data_type: data_type.clone(),
         }))
-        .collect()
 }
 
-fn unwrap_option_data_type(data_type: Type) -> Result<Type> {
+fn unwrap_option_type(data_type: Type) -> Type {
```

Comment:
Why do you now panic instead of returning errors?

Reply:
Returning errors here does not actually improve error messages.
We also don't have spans to the original source location anymore at this point, so we'd have to hand that all the way through. Alternatively, these checks could be duplicated to where the context structs are parsed and we just continue unwrapping here.
Ideally we wouldn't need any checks here because the data types would be designed in such a way that makes invalid states impossible to represent.